### PR TITLE
Make trials a kwarg in run_sim; make generate_trials mandatory

### PR DESCRIPTION
**Pull request by lrennels**

### CODE REVIEW

**Comment by lrennels** [comment]
Build failures are due to dependencies, which I am fixing separately and will then update the commit numbers.

File: docs/src/tutorials/tutorial_4.md [hunk 1]
```diff
@@ -102,12 +102,12 @@ generate_trials!(sim, 1000, filename="/tmp/trialdata.csv")
...
+Finally, use the [`set_models!`](@ref) and [`run_sim`](@ref) functions.  First, calling [`set_models!`] with a model, marginal model, or list of models will set those models as those to be run by your `sim` simulation.  Next, use [`run_sim`](@ref) which runs a simulation, with parameters describing the number of trials and optional callback functions to customize simulation behavior. In its simplest use, the [`run_sim`](@ref) function iterates over all pre-generated trials, perturbing a chosen set of Mimi's "external parameters", based on the defined distributions, and then runs the given Mimi model. Optionally, trial values and/or model results are saved to CSV files.  View the internals documentation for **critical and useful details on the full signature of this function**:
```
Comment (rjplevin) [inline docs/src/tutorials/tutorial_4.md]: I think "iterates over all pre-generated **trial data**" would be more precise.

File: src/mcs/montecarlo.jl [hunk 2]
```diff
@@ -340,7 +340,7 @@ end
             scenario_placement::ScenarioLoopPlacement=OUTER,
             scenario_args=nothing)
 
-Run the indicated the first `trials`, which indicates the number of trials to run
+Optionally run the indicated the first `trials`, which indicates the number of trials to run
```
Comment (rjplevin) [inline src/mcs/montecarlo.jl]: Typo: "run the indicated the first"

**Comment by lrennels** [comment]
@rjplevin  thanks! Changes made

**Pull request merged**